src/Inciter/Partitioner.h

```diff
@@ -1283,7 +1284,7 @@ class Partitioner : public CBase_Partitioner< HostProxy,
     //! Signal back to host that we are ready for partitioning the mesh
     void signal2host_setup_complete( const CProxy_Transporter& host ) {
       Group::contribute(
-        CkCallback(CkIndex_Transporter::redn_wrapper_partition(NULL), host ));
+        CkCallback(CkIndex_Transporter::redn_wrapper_part(NULL), host ));
```

Comment:
If we need to use a barrier because of our reliance on MPI, go for it. That was my question. I apologize if that was obvious, but I don't have the deep understanding of quinoa you do. In fact, most of my comments are to that effect. Hero programmers, regardless of what some people say, are a bad thing and should be avoided like the plague. Code should be understandable and easy to follow

So if the solution of a barrier makes sense, go for it. Sometimes you actually DO want a barrier (particularly when you need quiescence to change to MPI)

As for reductions versus p2p: Yes, if you are doing a pattern that is a reduction then it is hard to beat a reduction. And if you truly need one, go for it. Same with barriers. But if you can avoid the barrier, you should. Because even if you do need everyone to communicate in a tree pattern, you can potentially get your signal, send your signal, and then get back to work. The alternative is having to wait until everyone has received the signal and has the result (or dummy value if it is just a barrier) which involves a lot of idle time and is the graph of choice for runtime developers arguing the utility of a tool.

Maybe when MPI's non-blocking reductions are more ubiquitous that will become the gold standard as it combines the best of both worlds. But as it is right now, we need to weigh the pros and cons of everything. Hence my asking if this was the approach we should take.